### ISSUE TITLE
Brakes HATBM with Rails 4.2.0.rc2

### ISSUE THREAD

**Issue by JanStevens** [magnusvk/counter_culture#88]
Hello,

After a long and deep bug hunting I found out that this gem brakes HABTM relations in a very fun way:

```
Account should fail
     Failure/Error: Unable to find matching line from backtrace
     ActiveModel::MissingAttributeError:
       can't write unknown attribute ``
     # ./spec/support/database_cleaner.rb:60
```

I've create an example project on GitHub: https://github.com/JanStevens/counting_culture-issue4.2
Since we have a quite big stack I've included the typical gems (but Im confident the issue is related to counter culture). The app include Apartment, Postgres, Rspec, DatabaseCleaner and of course counter culture

First run the spec:
```
bundle exec rspec spec/models/account_spec.rb
```
It will fail

Then remove the gem in the `Gemfile` run the same spec again and TADA everything passes. As you can see my spec is very easy, I just create a HABTM relation object using FactoryGirl.

Notice that I DONT USE counter_culture in any of my example models I only have it listed in the Gemfile.

Regards

**Comment by adambarber** [comment]
I'm having the same problem as well.

* Rails 4.2.0
* counter_culture (0.1.30) from github

Removing counter_culture from the Gemfile causes tests to pass.

**Comment by jaygen** [comment]
:+1: Same issue here.

**Comment by magnusvk** [comment]
@JanStevens Thank you for providing the test project that reproduces this. I just spent a bunch of time diving into this to figure out what's going on. I was able to trace it to the after_comit_action gem -- a dependency of counter_culture. In fact, I've been able to fully remove both after_commit_action and counter_culture as dependencies and still reproduce this, see [this](https://github.com/JanStevens/counting_culture-issue4.2/pull/1).

At this point, I wonder if this is a Rails bug?

**Comment by magnusvk** [comment]
This is fixed in 0.1.31. Thanks for your patience, this was rather difficult to track down. In all honesty, I still don't fully understand what's going on but I was able to side-step it. I was able to confirm that this fixes it in your test project by simply running `bundle update counter_culture`.
